src/Type/Php/RegexCapturingGroup.php

```diff
@@ -5,49 +5,99 @@
 class RegexCapturingGroup
 {
 
+	private int $id;
+
+	private bool $forceNonOptional = false;
+
+	private static int $idGenerator = 100;
+
 	private function __construct(
 		private ?string $name,
-		private bool $inAlternation,
+		private ?int $alternationId,
 		private bool $inOptionalQuantification,
 		private RegexCapturingGroup|RegexNonCapturingGroup|null $parent,
 	)
 	{
+		$this->id = self::$idGenerator;
+		self::$idGenerator++;
```

Comment:
I don't like global state like that. This should be passed as a plain number from the outside and counted in the instance that creates objects of this class.